java/arcs/core/allocator/Allocator.kt

```diff
@@ -103,14 +111,16 @@ class Allocator(val hostRegistry: HostRegistry) {
      */
     private fun createStorageKey(
         arcId: ArcId,
-        idGenerator: Id.Generator
-    ): StorageKey = VolatileStorageKey(
-            arcId,
-            idGenerator.newChildId(arcId, "").toString()
+        storageKey: CreateableStorageKey,
+        type: Type
+    ): StorageKey =
+        CapabilitiesResolver(CapabilitiesResolver.StorageKeyOptions(arcId)).createStorageKey(
+            storageKey.capabilities,
+            type.toSchemaHash()
+        )?.childKeyForHandle(storageKey.nameFromManifest) ?: throw Exception(
```

Comment:
where is `nameFromManifest` coming from?
for `create` handles of long running arc, their storageKeys should already be set at build time. for `create` handles in ephemeral arcs, i think you'd need to generate a unique id here for the handle's storage key. am i missing anything?

Reply:
nameFromManifest is the recipe handle name, e.g.

recipe FooRecipe
  fooHandle: creates

nameFromManifest == "fooHandle"

startArcForPlan() right now always creates a unique ArcId, so the storageKeys should be unique. Long running arcs are not handled yet, since the Plan objects don't encode ArcIds or provide a way of referencing an existing Arc